ash-swapchain/src/lib.rs

```diff
@@ -0,0 +1,368 @@
+        physical_device: vk::PhysicalDevice,
+        device: &Device,
+        frames_in_flight: usize,
+        extent: vk::Extent2D,
+    ) -> Self {
+        Self {
+            options,
+
+            frames: (0..frames_in_flight)
+                .map(|_| unsafe {
+                    Frame {
+                        complete: device
+                            .create_fence(
+                                &vk::FenceCreateInfo::builder()
+                                    .flags(vk::FenceCreateFlags::SIGNALED),
+                                None,
+                            )
+                            .unwrap(),
+                        acquire: device
+                            .create_semaphore(&vk::SemaphoreCreateInfo::default(), None)
+                            .unwrap(),
+                        generation: 0,
+                    }
+                })
+                .collect(),
+            frame_index: 0,
+
+            surface,
+            physical_device,
+            handle: vk::SwapchainKHR::null(),
+            generation: 0,
+            images: Vec::new(),
+            extent,
+            format: vk::SurfaceFormatKHR::default(),
+            needs_rebuild: true,
+
+            old_swapchains: VecDeque::new(),
+        }
+    }
+
+    /// Destroy all swapchain resources. Must not be called while any frames are still in flight on
+    /// the GPU.
+    pub unsafe fn destroy(&mut self, device: &Device, swapchain_fn: &khr::Swapchain) {
+        for frame in &self.frames {
+            device.destroy_fence(frame.complete, None);
+            device.destroy_semaphore(frame.acquire, None);
+        }
+        if self.handle != vk::SwapchainKHR::null() {
+            swapchain_fn.destroy_swapchain(self.handle, None);
+        }
+        for &(swapchain, _) in &self.old_swapchains {
+            swapchain_fn.destroy_swapchain(swapchain, None);
+        }
+    }
+
+    /// Force the swapchain to be rebuilt on the next [`acquire`](Self::acquire) call, passing in
+    /// the surface's current size.
+    pub fn update(&mut self, extent: vk::Extent2D) {
+        self.extent = extent;
+        self.needs_rebuild = true;
+    }
+
+    /// Maximum number of frames that may be concurrently rendered
+    pub fn frames_in_flight(&self) -> usize {
+        self.frames.len()
+    }
+
+    /// Latest set of swapchain images, keyed by [`AcquiredImage::image_index`]
+    pub fn images(&self) -> &[vk::Image] {
+        &self.images
+    }
+
+    /// Format of images in [`images`](Self::images), and the color space that will be used to
+    /// present them
+    pub fn format(&self) -> vk::SurfaceFormatKHR {
+        self.format
+    }
+
+    /// Dimensions of images in [`images`](Self::images)
+    pub fn extent(&self) -> vk::Extent2D {
+        self.extent
+    }
+
+    /// Acquire a swapchain image
+    ///
+    /// Returns [`vk::Result::ERROR_OUT_OF_DATE_KHR`] iff the configured format or present modes
```

Comment:
```suggestion
    /// Returns [`vk::Result::ERROR_OUT_OF_DATE_KHR`] if the configured format or present modes
```

Reply:
`iff` is deliberately used in the technical sense here. Did I miss a case?

Reply:
I think you should spell out `if-and-only-if` in this case. It looks like a typo.